### PR TITLE
fix(server): resolve monorepo favicons and sanitize SVGs

### PR DESCRIPTION
**Pull request by Franz1241**
## What Changed

- Resolve project favicons from one-level workspaces under `apps/*`, `packages/*`, and `services/*`.
- Reuse the existing favicon candidates and source metadata discovery for each workspace.
- Preserve root-level favicon precedence and deterministic monorepo search order.
- Serve project SVG favicons using only the root `<svg>` document, excluding XML declarations, doctypes, and trailing content.
- Add focused coverage for monorepo lookup, precedence, SVG extraction, and signed asset resolution.

## Why

T3 Code only searched favicon locations relative to the selected project root. Monorepos commonly keep their application favicon under paths such as `apps/web/public/favicon.svg`, so those projects fell back to the generic folder icon.

Some otherwise valid SVG favicon files also contain XML declarations or doctypes that can prevent them from rendering reliably as sidebar image assets. Extracting the SVG document before serving it keeps the original file untouched while avoiding that failure mode.

## UI Changes

No layout or component changes. Projects that previously displayed the fallback folder icon can now display their application favicon when it lives in a supported monorepo workspace.

Verified in an isolated local T3 Code environment with an XML/doctype-prefixed favicon at `apps/dashboard/public/favicon.svg`; both sidebar image instances loaded successfully at their intrinsic 150×150 size.

## Checklist

- [x] This PR is small and focused
- [x] I explained what changed and why
- [ ] I included before/after screenshots for any UI changes
- [x] I included a video for animation/interaction changes (not applicable)

---

> [!NOTE]
> **Medium Risk**
> Touches signed asset resolution and serves transformed SVG content (mitigates trailing markup) rather than always streaming the on-disk file; favicon precedence behavior changes for monorepos.
> 
> **Overview**
> **Favicon discovery** now walks one-level workspaces under `apps/`, `packages/`, and `services/` (sorted entries), reusing the same conventional filenames and HTML/metadata icon href resolution per app. Root `t3.json`, root candidates, and root source metadata still win before any monorepo path; unreadable monorepo directories are skipped.
> 
> **SVG project favicons** are no longer served as raw files when possible: `resolveAsset` reads `.svg` favicons, runs new `extractSvgDocument` to keep only the first root `<svg>` (drops XML/doctype/comments and trailing junk like extra `<script>`/second roots), and returns `{ kind: "text", contentType: "image/svg+xml" }`. Non-SVG favicons and unparseable SVGs still use file paths. The asset route serves the `text` variant with the same private cache and `nosniff` headers as file responses.
> 
> Tests cover SVG extraction edge cases, sanitized favicon resolution, and monorepo precedence/ordering.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 917649a8aba61ea5e84729dbaae91bb110d8590d. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

> [!NOTE]
> ### Fix favicon resolution for monorepos and sanitize SVG favicons
> - Extends `ProjectFaviconResolver.resolvePath` to search monorepo subdirectories (`apps`, `packages`, `services`) for favicons, checking well-known candidate paths and `<link rel="icon">` hrefs in source files, in stable sorted order.
> - Adds `extractSvgDocument` in [AssetAccess.ts](https://github.com/pingdotgg/t3code/pull/4424/files#diff-7106051c89d4fb8e7cebeee854e81bb55fdf03ec154089eb8c7499ef78ae51c1) to extract and sanitize the first complete `<svg>` element from raw file content, handling nested SVGs, comments, and CDATA.
> - When a `.svg` favicon is resolved, the server now reads the file, extracts the SVG document, and returns it as an inline text asset with `content-type: image/svg+xml` instead of serving the raw file.
> - Updates the asset HTTP route in [http.ts](https://github.com/pingdotgg/t3code/pull/4424/files#diff-55fb49cf72ecef123d8f2fa7061526dadb37cf93f8e9dd9e7f0d4f8cd4ff9f6d) to handle the new inline text asset variant with proper `Cache-Control` and `X-Content-Type-Options` headers.
> - Behavioral Change: callers requesting `.svg` favicons now receive sanitized inline SVG text; non-SVG favicons and failed extractions continue to return file assets.
>
> 
>
> <sup><a href="https://app.macroscope.com">Macroscope</a> summarized 917649a.</sup>
> 
>

### CODE REVIEW

File: apps/server/src/assets/AssetAccess.ts [hunk 1]
```diff
@@ -91,7 +91,22 @@ const AssetClaimsJson = Schema.fromJsonString(AssetClaimsSchema);
...
+    };
+
+export function extractSvgDocument(source: string): string | null {
+  const documentMatch = source.match(/<svg\b[\s\S]*<\/svg\s*>/i);
+  if (documentMatch) {
+    return documentMatch[0];
+  }
+  const selfClosingMatch = source.match(/<svg\b[^>]*\/\s*>/i);
+  return selfClosingMatch?.[0] ?? null;
+}
```
[the remark this answers is not in the document]
Comment (Franz1241) [reply apps/server/src/assets/AssetAccess.ts:178]: Fixed in 9c263aee7. SVG extraction now stops at the first closing root, with a regression test covering multiple SVG roots separated by script content.

File: apps/server/src/project/ProjectFaviconResolver.ts [hunk 2]
```diff
@@ -233,12 +213,106 @@ export const make = Effect.gen(function* () {
...
+          Effect.mapError(
+            (cause) =>
+              new ProjectFaviconResolutionError({
+                operation: "read-monorepo-root",
+                workspaceRoot: projectCwd,
+                relativePath: monorepoRoot,
+                absolutePath: absoluteRoot,
+                cause,
+              }),
+          ),
```
[the remark this answers is not in the document]
Comment (Franz1241) [reply apps/server/src/project/ProjectFaviconResolver.ts]: Fixed in 9c263aee7. Each monorepo collection is now a best-effort probe: stat or readDirectory failures skip that root and resolution continues to later collections. Added a regression test covering both failure modes before a valid services favicon.

File: apps/server/src/assets/AssetAccess.ts [hunk 3]
```diff
@@ -91,7 +91,22 @@ const AssetClaimsJson = Schema.fromJsonString(AssetClaimsSchema);
...
+export type ResolvedAsset =
+  | { readonly kind: "file"; readonly path: string }
+  | {
+      readonly kind: "text";
+      readonly body: string;
+      readonly contentType: "image/svg+xml";
+    };
+
+export function extractSvgDocument(source: string): string | null {
+  const documentMatch = source.match(/<svg\b[\s\S]*?<\/svg\s*>/i);
```
[the remark this answers is not in the document]
Comment (Franz1241) [reply apps/server/src/assets/AssetAccess.ts]: Fixed in 917649a8a. Regex extraction was replaced with a quote-aware tag scanner that balances nested SVG elements while returning only the first root document. Added a nested SVG regression test.

**Comment by juliusmarminge** [comment]
Closing as part of the open-PR backlog sweep (wave 1).

**Reason:** Jul conflicted monorepo favicon/SVG work; favicon caching landed in #4767.
**Related:** #4767

Reopen if this is still wanted and you’re willing to rebase onto current `main`.
